src/ufuncs.c

```diff
@@ -223,6 +230,35 @@ static void Y_Y_func(char **args, npy_intp *dimensions,
 }
 static PyUFuncGenericFunction Y_Y_funcs[1] = {&Y_Y_func};
 
+/* Define the geom -> prepared geom functions (Y_Yp) */
+static void *prepare_data[1] = {GEOSPrepare_r};
+typedef void *FuncGEOS_Y_Yp(void *context, void *a);
+static char Y_Yp_dtypes[2] = {NPY_OBJECT, NPY_OBJECT};
+static void Y_Yp_func(char **args, npy_intp *dimensions,
+                      npy_intp *steps, void *data)
+{
+    FuncGEOS_Y_Yp *func = (FuncGEOS_Y_Yp *)data;
+    void *context_handle = geos_context[0];
+    GEOSGeometry *in1;
+    GEOSPreparedGeometry *ret_ptr;
+
+    UNARY_LOOP {
+        /* get the geometry: return on error */
+        if (!get_geom(*(GeometryObject **)ip1, &in1)) { return; }
+
+        if (in1 == NULL) {
+            /* in case of a missing value: return NULL (NaG) */
+            ret_ptr = NULL;
+        } else {
+            ret_ptr = func(context_handle, in1);
```

Comment:
This is currently crashing (on Mac, GEOS 3.8) with `terminated by signal SIGSEGV (Address boundary error)`

I think the way I've wrapped the `GEOSPrepare` function used here seems valid - it copies how this was done for other unary ufuncs above.

@caspervdw I'm probably missing something really obvious here that you're likely to spot quickly - are you able to point me in the right direction?

Reply:
I also get a segfault, but it seems it is not crashing here, but in `PreparedGeometryObject_FromGEOSPreparedGeometry` (called in `OUTPUT_Yp` below). And there, it seems the `tp_alloc` call segfaults. But not directly sure why this happens, since the PreparedGeometryObject seems very similar to GeometryObject.

Reply:
Sorry for my late response, I have been very busy.

I don't see anything obvious going wrong, but I am thinking whether the `GEOSPreparedGeometry` is able to function when the related GEOSGeometry is already deallocated? Could a segfault be originating from that issue?